src/dbnode/storage/namespace.go

```diff
@@ -875,9 +875,26 @@ func (n *dbNamespace) Bootstrap(start time.Time, process bootstrap.Process) erro
 			wg.Done()
 		})
 	}
-
 	wg.Wait()
 
+	retrieverMgr := n.opts.DatabaseBlockRetrieverManager()
+	// May be nil depending on the caching policy.
+	if retrieverMgr != nil {
+		// Attempt to call CacheShardIndices now that all the shards are bootstrapped. Cannot call
+		// this earlier as block lease verification will fail due to the shards not being bootstrapped
+		// (and as a result no leases can be verified since the flush state is not yet known).
+		retriever, err := n.opts.DatabaseBlockRetrieverManager().Retriever(metadata)
+		if err != nil {
+			multiErr = multiErr.Add(err)
+		} else {
+			shardIDs := make([]uint32, 0, len(shards))
+			for _, shard := range shards {
+				shardIDs = append(shardIDs, shard.ID())
+			}
+			retriever.CacheShardIndices(shardIDs)
```

Comment:
Can we add a log just before this statement and just after?  This takes a long time sometimes and if we don't have the log anymore it will hurt operational visibility.

Reply:
yes thank you for the reminder, I had meant to do it but forgot. Will add it now